arangod/IResearch/IResearchView.cpp

```diff
@@ -1613,21 +1622,29 @@ arangodb::Result IResearchView::link(
   dataPath /= subPath;
   dataPath /= arangodb::iresearch::IResearchView::type().name();
   dataPath += "-";
-  dataPath += std::to_string(view->id());
+  dataPath += std::to_string(id);
+
+  auto view = std::unique_ptr<IResearchView>(
+    new IResearchView(&vocbase, info, std::move(dataPath), isNew)
+  );
+
+  auto props = info.get("properties");
+
+  if (props.isNone()) {
+    // if no 'properties' then assume defaults
+    props = emptyObjectSlice();
+  }
 
-  PTR_NAMED(IResearchView, ptr, view, info, std::move(dataPath));
-  auto& impl = reinterpret_cast<IResearchView&>(*ptr);
-  auto& json = info.isNone() ? emptyObjectSlice() : info; // if no 'info' then assume defaults
   std::string error;
 
-  if (!impl._meta.init(json, error)) {
+  if (!view->_meta.init(props, error)) {
     LOG_TOPIC(WARN, iresearch::IResearchFeature::IRESEARCH)
       << "failed to initialize iResearch view from definition, error: " << error;
 
     return nullptr;
   }
 
-  return std::move(ptr);
+  return std::shared_ptr<IResearchView>(std::move(view));
```

Comment:
If we move to a `shared_ptr` here, we might as well create a `shared_ptr` right away above and then use return value optimization.

Reply:
yes, let me just use `shared_ptr` above instead of `unique_ptr` and simply return it